fastify.js

```diff
@@ -621,7 +621,7 @@ function fastify (options) {
     // https://github.com/nodejs/node/blob/6ca23d7846cb47e84fd344543e394e50938540be/lib/_http_server.js#L666
 
     // If the socket is not writable, there is no reason to try to send data.
-    if (socket.writable && socket.bytesWritten === 0) {
+    if (socket.writable && socket.write) {
```

Comment:
```suggestion
    if (socket.writable) {
```

`socket.write` will always evaluate to true, we can remove it completely

Reply:
It was failing in one of the tests.

Reply:
```
​ FAIL ​ test/request-error.test.js
 ✖ socket.write is not a function

    if (socket.writable) {
      socket.write(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\nContent-Type: application/json\r\n\r\n${body}`)
-------------^
    }
    socket.destroy(err)

  test: default clientError handler destroys sockets in writable state
  stack: |
    Object.defaultClientErrorHandler (fastify.js:625:14)
    Test.<anonymous> (test/request-error.test.js:160:18)
  at:
    line: 625
    column: 14
    file: fastify.js
    function: Object.defaultClientErrorHandler
  type: TypeError
  tapCaught: testFunctionThrow

​ FAIL ​ test/request-error.test.js 1 failed of 30 901.784ms
 ✖ socket.write is not a function

Suites:   ​1 failed​, ​1 of 1 completed​
Asserts:  ​​​1 failed​, ​29 passed​, ​of 30​
code: 1
signal: null
TAP>
```

I am tweaking the test